omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java

```diff
@@ -84,4 +95,32 @@ public void postAttachment_shouldUpdateObsComment() throws Exception {
 		String comment = (String) PropertyUtils.getProperty(doc, "comment");
 		assertEquals(editedComment, comment);
 	}
+	
+	@Test
+	public void doSearch_shouldReturnResult() {
+		
+		// Setup
+		AttachmentResource1_10 res = new AttachmentResource1_10();
+		Patient patient = Context.getPatientService().getPatient(2);
+		
+		AttachmentsService as = Context.getService(AttachmentsService.class);
+		List<Attachment> attachments = as.getAttachments(patient, false);
+		
+		// Replay
+		MockHttpServletRequest request = newGetRequest(getURI() + "/");
+		request.addParameter("patient", patient.getUuid());
+		request.addParameter("v", "full");
+		
+		RequestContext requestContext = new RequestContext();
+		requestContext.setRequest(request);
+		PageableResult response = res.doSearch(requestContext);
+		
+		BasePageableResult basePageableResult = (BasePageableResult) response;
```

Comment:
That's a detail but if possible: `basePageableResult` → `pageableResult`.

Reply:
pageableResult doesn't return a list of results whereas BasePageableResult do, however an improvement is possible